connectors/sources/outlook.py

```diff
@@ -456,6 +473,57 @@ async def get_user_accounts(self):
                 yield user_account
 
 
+class MultiOffice365Users(BaseOffice365User):
+    """Fetch multiple Office365 users based on a list of email addresses."""
+
+    def __init__(self, client_id, client_secret, tenant_id, client_emails: List[str]):
+        super().__init__(client_id, client_secret, tenant_id)
+        self.client_emails = client_emails
+
+    async def get_users(self):
+        access_token = await self._fetch_token()
+        for email in self.client_emails:
+            url = f"https://graph.microsoft.com/v1.0/users/{email}"
```

Comment:
I imagine that this is not particularly efficient if you have a large list of client emails. Did you look to see if there's a way to fetch them by email address in bulk?

Reply:
104e299
I updated the implementation to use JSON batching which processes client emails in batches of up to 20 (max supported by Graph API). I also handled exceptions and collected errors which are raised after processing all batches.

[Graph API JSON batching Documentation](https://learn.microsoft.com/en-us/graph/json-batching)